### DISCUSSION TITLE
Set a minimum nr of time steps to disable a transition from one state to another

### DISCUSSION THREAD

**Discussion by NunoSH** [bmcclintock/momentuHMM#105] [Q&A] [answered]
Hi!
is there a way to set a minimum of consecutive time-steps (nr of GPS points) as a constrain to disable the transition from one state to another?

I know the states that I am hoping to predict, they are of a certain length (more than 250m) and they do not occur on short time-steps or distances. But sometimes the tracks on a state i, for a brief moment, behaves as state j (step length, angle, etc). So I would like to eliminate those "fake" transitions based on a minimum distance (or time steps).

State i to state j they do can transit to one another, but these states are not of short legth or time period.

Is there any way to tackle this issue?

I hope I made myself clear :)

Thank you very much!
 Nuno

**Comment by bmcclintock** [comment, marked as answer]
I'm not sure I understand exactly what constraint you're trying to impose, but if the basis of the constraint is something that can be pre-calculated and included as a covariate in the data, then this covariate could be used to enforce the constraint on the state transition probabilities using the `formula` and `fixPar` arguments. See #68 for an example.

**Reply by NunoSH** [reply]
Hi Brett,
first, Thanks a lot for you swift reply! 
well, it doesn't has to necessarily has to be a constrain (Im not sure why I wrote that). 
What I was wondering if if there is a way of telling the model that " it can only be state "traveling" if there are at least X consecutive GPS points predicted as state "traveling". 

I guess one way of looking at it is to ask the model to reduce its sensitivity to sudden changes in behaviour within a longer sequence of homogeneous behaviour.   

Now that I think of it, I have my data interpolated at 1min time-step. Maybe if the data points are a bit more sparse, it could help on this....

Anyway, Thank you for the example. I will have a deeper look at it!

Thanks again Brett and thank you for MomentuHMM! It is great!
Cheers,
Nuno
